GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -225,16 +225,42 @@ protected override void Dispose(bool disposing)
         {
             if (disposing)
             {
+                //// MenuCommands not disposable
+                //// _superprojectCurrentCheckout not disposable
+                //// _selectedObjectIds not disposable
+                //// _baseCommitToCompare not disposable
                 _revisionSubscription?.Dispose();
                 _revisionReader?.Dispose();
+                //// _artificialStatus not disposable
+                //// _ambiguousRefs not disposable
+                //// _refFilterOptions not disposable
+                _lastVisibleResizableColumn?.Dispose();
+                _maximizedColumn?.Dispose();
+                //// _revisionGraphColumnProvider not disposable
+                _selectionTimer?.Dispose();
                 _buildServerWatcher?.Dispose();
 
                 if (_indexWatcher.IsValueCreated)
                 {
                     _indexWatcher.Value.Dispose();
                 }
 
+                //// _authorHighlighting not disposable
+                //// _parentChildNavigationHistory not disposable
+                //// _quickSearchProvider not disposable
+                //// _toolTipProvider  not disposable
+                _loadingControlSync.Dispose();
+                _loadingControlAsync.Dispose();
```

Comment:
❕ Both `_loadingControlSync` and `_loadingControlAsync` get added to the `Controls` collection. When present in this collection, they are automatically disposed, and should not be disposed again here. The ideal solution to this is relocating the code that adds these to `Controls` from inside of `ShowLoading` to the constructor, which ensures there is never a need to dispose these controls manually.

Reply:
> Both `_loadingControlSync` and `_loadingControlAsync` get added to the `Controls` collection

Yes, though not by the ctor but by `ShowLoading` from `OnCreate` amongst others.
BTW: `ShowLoading` is called on every refresh and adds these controls yet again.